src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php

```diff
@@ -160,7 +160,9 @@ protected static function ensureKernelShutdown()
             static::$kernel->shutdown();
             static::$booted = false;
 
-            if ($container instanceof ResetInterface) {
+            if ($container->has('services_resetter')) {
```

Comment:
Does the added rest really reach this part? The container still implements the `ResetInterface`, doesn’t it?

Reply:
Sorry, I'm not sure I understand the question.

Reply:
Let me rephrase it. I would expect the added test to pass even without the change here.

Reply:
But it doesn't. `ResettableService` doesn't implement `ResetInterface`, which is how the `Container::reset()` method works: https://github.com/symfony/symfony/blob/fea8f98676e705d696307708f599ad096c633406/src/Symfony/Component/DependencyInjection/Container.php#L299-L313

The `services_resetter` on the other hand takes into account services without the interface:  
https://github.com/symfony/symfony/blob/fea8f98676e705d696307708f599ad096c633406/src/Symfony/Component/HttpKernel/DependencyInjection/ServicesResetter.php#L39-L50

Reply:
thanks for the clarification 👍